client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,254 @@
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    profiles = []
+
+    def process(self, instance):
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        job_info = AYONDeadlineJobInfo.from_dict(attr_values)
+
+        self._handle_machine_list(attr_values, job_info)
+
+        self._handle_additional_jobinfo(attr_values, job_info)
+
+        instance.data["deadline"]["job_info"] = job_info
+
+        # pass through explicitly key and values for PluginInfo
+        plugin_info_data = json.loads(attr_values["additional_plugin_info"])
+        instance.data["deadline"]["plugin_info_data"] = plugin_info_data
+
+    def _handle_additional_jobinfo(self,attr_values, job_info):
+        """Adds not explicitly implemented fields by values from Settings."""
+        additional_job_info = attr_values["additional_job_info"]
+        for key, value in json.loads(additional_job_info).items():
+            setattr(job_info, key, value)
+
+    def _handle_machine_list(self, attr_values, job_info):
+        machine_list = attr_values["machine_list"]
+        if machine_list:
+            if job_info.MachineListDeny:
+                job_info.Blacklist = machine_list
+            else:
+                job_info.Whitelist = machine_list
+
+    @classmethod
+    def apply_settings(cls, project_settings):
+        settings = project_settings["deadline"]
+        profiles = settings["publish"][cls.__name__]["profiles"]
+
+        cls.profiles = profiles or []
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        task_entity = create_context.get_task_entity(folder_path, task_name)
+
+        task_name = task_type = None
+        if task_entity:
+            task_name = task_entity["name"]
+            task_type = task_entity["taskType"]
+        profile = filter_profiles(
+            cls.profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_type,
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        if not profile:
+            return []
+        overrides = set(profile["overrides"])
+        if not overrides:
+            return []
+
+        defs = [
+            UISeparatorDef("deadline_defs_starts"),
+        ]
+
+        defs.extend(cls._get_artist_overrides(overrides, profile))
+
+        # explicit
+        defs.append(
+            TextDef(
+                "frames",
+                label="Frames",
+                default="",
+                tooltip="Explicit frames to be rendered. (1, 3-4)"
+            )
+        )
+
+        defs = cls._host_specific_attr_defs(create_context, instance, defs)
```

Comment:
With access to the instance and the create context this can be WAY more specific than just `host_specific` so I would name it different like `_additional_attr_defs` and then clearly label the method that it's about inherited subclasses, like e.g. host specific implementations.

Reply:
Had to change it to this instead of original subclasses as they didn't work (they were doubling attr defs). Its ugly, but its working.

Reply:
I'm fine with it exposed here like this - I'm just saying we should rename the method name.